pkg/auth/agent_local_authenticator.go

```diff
@@ -66,6 +68,20 @@ func (a *AgentLocalAuthenticator) EnableOrgBasedFeatureGates() bool {
 }
 
 func (a *AgentLocalAuthenticator) AuthAgentAuth(token string) (interface{}, error) {
+	// Verify is token is expired or not in case of add nodes workflow
+	if a.WorkflowType == AgentWorkflowTypeAddNodes {
```

Comment:
I wonder if the check can be simplified by removing the WorkflowType check. The token contains the expiration time only in the addnodes workflow. So if the expiration time exist, then check if it has expired. If it doesn't have the expiration time, then it is the install workflow and the expiration time check is skipped.

Reply:
`ParseExpirationFromToken` is an internal function of assisted and is used in other cases. If we always parse the token for the expiration time then in the case of install workflow, the existing check in the `ParseExpirationFromToken` will fail and error with `token missing 'exp' claim` https://github.com/openshift/assisted-service/blob/8703b59882a2f9d5730ece7c2c47cb025ab42a8f/internal/gencrypto/token.go#L107-L110 i.e. it will incorrectly expect the exp claim in the case of install workflow hence the check on workflowType